examples/compiler_projects/Dev-C++/4_2_Pwm/config_header__4_2_Pwm-doc.txt


 \section PrjConfig4_2_Pwm List of configuration settings for 4_2_Pwm
 This is only a copy with doxygen ready comment blocks from the original file ../../../../examples/src/Tutorials/4_SupplementaryDriver//4_2_Pwm/config_4_2_Pwm.h 
 This header is included by <IsoAgLib/library/xgpl_src/IsoAgLib/isoaglib_config.h> based on the
 project define #define PRJ_USE_AUTOGEN_CONFIG config_4_2_Pwm.h ( Important: place the directory of the application source files in the include search path )

 Use the file conf_4_2_Pwm in the directory /home/bf/osb/VtServer_StandardRevision3_ActiveWork/IsoAgLib/examples/compiler_projects/Dev-C++/ as input file for ./update_makefile.sh to create the project generation files.
\code
// File: config_4_2_Pwm.h
// IMPORTANT: Never change the first block of this header manually!!!
//            All manual changes are overwritten by the next call of "update_makefile.sh conf_4_2_Pwm" 
//            Perform changes direct in the feature and project setup file conf_4_2_Pwm
//  ALLOWED ADAPTATION: Move the to be adapted defines from the middle block to the end after
//                      the line START_INDIVIDUAL_PROJECT_CONFIG and remove the comment indication there.
//                      All commented out defines in the middle block will be upated on next "update_makefile.sh conf_4_2_Pwm" call,
//                      if the corresponding value in isoaglib_config.h changed


// include an external file for definition of pool and firmware versions
#include "version.h"


#ifndef SYSTEM_PC
#define SYSTEM_PC
#endif // SYSTEM_PC
#define PRJ_USE_AUTOGEN_CONFIG config_4_2_Pwm.h


#define CAN_BUS_CNT 1 

#define CAN_BUS_USED 0 

#define CAN_INSTANCE_CNT 1 

#define PRT_INSTANCE_CNT 1 

#define RS232_INSTANCE_CNT 1 

// Decide if the CPU stores number variables in BIG or LITTLE endian byte order in memory.
// Most CPU will use LITTLE ENDIAN. Only some types of ARM, mostly 68k and PowerPC CPU types will use big endian.
// Please check the manual of your targret cpu. This setting is used to activate some quick number conversion algorithms,
// which provide quick conversion from number variable to CAN strings ( which are always little endian in ISO ) - and other way.
#define OPTIMIZE_NUMBER_CONVERSIONS_FOR_LITTLE_ENDIAN

// #define OPTIMIZE_NUMBER_CONVERSIONS_FOR_BIG_ENDIAN

// #define USE_PCAN_LIB

// #define CONFIG_DO_NOT_START_RELAIS_ON_STARTUP

// Decide if HEAP allocation strategy shall reduce size about 5K to 10K in favour of speed
// Strong Advice: Don't activate this, as long your target has not too tight memory restrictions
// Initialization of CAN filters and of local process data might get too slow under worst case conditions
// #define OPTIMIZE_HEAPSIZE_IN_FAVOR_OF_SPEED

/** allow configuration by parameter value YES */
#ifndef YES
  #define YES 1
#endif
/** allow configuration by parameter value NO */
#ifndef NO
  #define NO 0
#endif
#ifndef USE_PROCESS_YN 
	#define USE_PROCESS_YN NO 
#endif
#ifndef USE_EEPROM_IO_YN 
	#define USE_EEPROM_IO_YN NO 
#endif
#ifndef USE_DATASTREAMS_IO_YN 
	#define USE_DATASTREAMS_IO_YN NO 
#endif
#ifndef USE_ISO_11783 
	#define USE_ISO_11783 
#endif

// The following configuration values can be overwritten.
// These settings are initially defined in isoaglib_config.h .
// These settings are in commented-out, so that you can activate and adapt them by
// moving them below the line with START_INDIVIDUAL_PROJECT_CONFIG

/// select reaction on powerdown detection - can be manually overridden in project config file
// #define CONFIG_DEFAULT_POWERDOWN_STRATEGY IsoAgLib::PowerdownByExplcitCall

/// define consolidation time for CAN_EN loss detection - can be manually overridden in project config file
// #define CONFIG_BUFFER_SHORT_CAN_EN_LOSS_MSEC 1000

/// set puffer size for CAN send
// #define CONFIG_CAN_SEND_BUFFER_SIZE 20

/// set default puffer size for CAN receive
// #define CONFIG_CAN_STD_LOAD_REC_BUF_SIZE_MIN 15

/// high load of some CAN idents need larger buffers
// #define CONFIG_CAN_HIGH_LOAD_REC_BUF_SIZE_MIN 25

/// set increment for automatic buffer increase after overload ( 0 == no increase )
// #define CONFIG_CAN_BUF_INCREASE_ON_OVERLOAD 5

/// define list of CAN filters where large buffers are needed due to high load
// #define CONFIG_CAN_HIGH_LOAD_IDENT_LIST { 0xc70000, 0xc80000, 0xcb0000, 0xeb0000, 0xec0000  }

/// define amount of list entries in CONFIG_CAN_HIGH_LOAD_IDENT_LIST
// #define CONFIG_CAN_HIGH_LOAD_IDENT_CNT 5

/// define mask for compare with ident
// #define CONFIG_CAN_HIGH_LOAD_IDENT_MASK 0xff0000

/// Multiply this SizeInPackets by 7 to get the size of ONE CHUNK (ONLY OF INTEREST IF STREAM IS CHUNKED)
// #define CONFIG_MULTI_RECEIVE_CHUNK_SIZE_IN_PACKETS 16

/// Maximum amount of packets to be allowed to be CTS'd in parallel (this amount will be distributed among all running streams)
// #define CONFIG_MULTI_RECEIVE_MAX_OVERALL_PACKETS_ADDED_FROM_ALL_BURSTS 16

/// Maximum amount of packets to be allowed by a "CTS" message.
// #define CONFIG_MULTI_RECEIVE_MAX_PER_CLIENT_BURST_IN_PACKETS CONFIG_MULTI_RECEIVE_MAX_OVERALL_PACKETS_ADDED_FROM_ALL_BURSTS

/// defines the amount of msec. to be wait until sending out the next CTS to the sender IF ONE STREAM ONLY BEING RECEIVED
// #define CONFIG_MULTI_RECEIVE_CTS_DELAY_AT_SINGLE_STREAM 0

/// defines the amount of msec. to be wait until sending out the next CTS to the sender IF MULTIPLE STREAMS BEING RECEIVED
// #define CONFIG_MULTI_RECEIVE_CTS_DELAY_AT_MULTI_STREAMS 50

/// define default RS232 baudrate - can be manually overridden in project config file
// #define CONFIG_RS232_DEFAULT_BAUDRATE 19200

/// define default RS232 bit encoding - can be manually overridden in project config file
// #define CONFIG_RS232_DEFAULT_DATA_MODE RS232IO_c::_8_N_1

/// define default RS232 XON-XOFF usage - can be manually overridden in project config file
// #define CONFIG_RS232_DEFAULT_XON_XOFF false

/// define default size of RS232 send puffer - can be manually overridden in project config file
// #define CONFIG_RS232_DEFAULT_SND_PUF_SIZE 200

/// define size of RS232 receive puffer - can be manually overridden in project config file
// #define CONFIG_RS232_DEFAULT_REC_PUF_SIZE 10

/// set default PWM frequency for digital output - can be manually overridden in project config file
// #define CONFIG_PWM_DEFAULT_FREQUENCY 100000

/// define size of buffer for Hdd read access - can be manually overridden in project config file
// #define CONFIG_HDD_READ_PUFFER_SIZE 100

/// define minimum size of data to read as one bock from Hdd - can be manually overridden in project config file
// #define CONFIG_HDD_READ_BLOCK_SIZE 20

/// define longest token which is used to translate printed numbers to number variables - can be manually overridden in project config file
// #define CONFIG_HDD_MAX_TOKEN_LEN 10

/// define value coding for RS232 connected DOS Drive - can be manually overridden in project config file
// #define CONFIG_HDD_DEFAULT_DATA_MODE RS232IO_c::_8_N_1

/// default max execution time of iScheduler_c::timeEvent - can be manually overridden in project config file
// #define CONFIG_DEFAULT_MAX_SCHEDULER_TIME_EVENT_TIME 500

/// erase ISO 11783 items after time (>0), on missing address claim after request - can be manually overridden in project config file
// #define CONFIG_ISO_ITEM_MAX_AGE 3000

/// interval [ms] for value update in EEPROM for local process data - can be manually overridden in project config file
// #define CONFIG_PROC_STORE_EEPROM_INTERVAL 5000

/// CAN BUS number for IsoAgLib (0xFF forces explicit call of init, to open the CAN BUS )
// #define CONFIG_CAN_DEFAULT_BUS_NUMBER 0xFF

/// define default MsgObj_c number to use for CAN send
// #define CONFIG_CAN_DEFAULT_SEND_OBJ 0

/// define default min number for CAN Msg-Obj used for IsoAgLib
// #define CONFIG_CAN_DEFAULT_MIN_OBJ_NR 0

/// define default max number for CAN Msg-Obj used for IsoAgLib
// #define CONFIG_CAN_DEFAULT_MAX_OBJ_NR 13

/// max time intervall of CAN err before switching back from 250 to 125 kbit/s in [msec.]
// #define CONFIG_CAN_MAX_CAN_ERR_TIME_BEFORE_SLOWERING 3000

/// define bitrate to use after fallbcack /ref CONFIG_CAN_MAX_CAN_ERR_TIME_BEFORE_SLOWERING
// #define CONFIG_CAN_FALLBACK_BITRATE 125

/// BUS OFF if CAN send lasts longer then defined
// #define CONFIG_CAN_MAX_SEND_WAIT_TIME 200

/// BUS problem, if no message received in this time
// #define CONFIG_CAN_MAX_CAN_IDLE 3000

/// YES | NO for EEPROM editor access via CAN
// #define CONFIG_EEPROM_USE_CAN_EDITOR_YN NO

/// YES | NO for EEPROM editor access via RS232
// #define CONFIG_EEPROM_USE_RS232_EDITOR_YN NO

/// set CAN Ident_c for filtering out EEEditor msgs
// #define CONFIG_EEPROM_USE_CAN_REC_IDENT 0x700

/** set CAN Msg obejct for receiving EEEditor msg */
// #define CONFIG_EEPROM_USE_CAN_REC_OBJ   13

/** set CAN Msg obejct for sending EEEditor msg */
// #define CONFIG_EEPROM_USE_CAN_SEND_OBJ  14

/** set CAN BUS number for CAN EEPROM Editor */
// #define CONFIG_EEPROM_USE_CAN_BUS       0

/** set CAN puffer size for receive and send */
// #define CONFIG_EEPROM_USE_CAN_BUFFER_SIZE    10

/** to use standard ident (11bit) ->0, 29bit -> 1 */
// #define CONFIG_EEPROM_USE_CAN_EXT_IDENT 0


// DONT REMOVE THIS AND THE FOLLOWING LINE AS THEY ARE NEEDED TO DETECT YOUR PERSONAL PROJECT ADAPTATIONS!!!
// START_INDIVIDUAL_PROJECT_CONFIG
\endcode


changelog.txt
/*! \page PageChanges Changelog of IsoAgLib


INCLUDE_doc/txt/mainHeaderWithNavbar.txt_INCLUDE

<table class="infobox" border="0">
<tr><th>Section Overview</th></tr>
<tr><td>The following parts are described on the respective pages:
<ol type="I">
    <li>\ref PageDeveloperInfo "Developer Information Page"
        <ul><li> \ref WhereToStart 
    <ul><li> \ref StartToLearn
        <li> \ref Documentation </li>
        <li> \ref GettingHelp </li>
    </ul>
    </li>
        <li> \ref MemoryUsage </li>
        </ul>
    </li>
    <li> Changelog </li>
    <li> \ref PageTodo "Main TODO Items Page"</li>
</ol>
</td></tr>
</table>
<div style="clear:left"></div>

\section Changes Changes

\subsection Change_O 2.0 to 2.1
  - Process data: GeneralCommand_c renamed to ProcessCmd_c
  - new naming scheme (m-prefix for member-variables, ...)
  - ProcIdent_c and all derived classes (local and remote): parameter c_ownerISOName removed from init() and constructor
  - changed handling of received process data messages (first local then remote check)
  - usage of fast datatypes in cases where an integer variable only needs a specific minimum size (HANDLE_FAST_DATATYPE_AS_STRICT)
  - optional global mutex in Scheduler_c so that an application can have several parallel threads with access to the IsoAgLib data structures
  - the IsoAgLib core files ( "xgpl_src" "commercial_BIOS" and simulated_IO") have been moved together in a new folder "library"
  - some important folders and files have been renamed so the content should be easier to identify
    - folders below IsoAgLib/comm are now labeled like the related OSI layer
    - IsoAgLib_Examples -> examples with a new folder "src" inside containing the source files
    - Binaries like vt2iso and proc2ito are now being compiled to the new folder "bin".
    - compiler_projects has been moved into "examples" containing the config files of the examples.
    - The folder "build_utils" has been renamed to "build" and moved to library/xgpl_src/build
    - The class "ElementBase_c" has been renamed to "SchedulerTask_c" and moved to folder "Scheduler", like "CanPkExt_c" as well.
    - The classes "CanCustomer_c" and "CanPkg_C" have been moved to folder "/driver/can"
  - optimized scheduler resorting to avoid several HEAP malloc/free calls
  - added new DEFINEs for MultiReceive: refer to new generated config_xxx.h or isoaglib_config.h
  - vt2iso: special parsing of proprietary elements and attributes possible
  - bug fix in Scheduler_c::resortTaskList(): task list resorting corrected
  - added language-auto-detection to vt2iso
  - added support for Auxiliary Assignment and passing the AuxInputStatus message to the application
  - added functionality to Base-Classes to suppress PGNs using a bitmask
  - bug fix: Address-Claim now also sent out after the 250ms-period if requested in this 250ms-period
  - introduce new target hardware: AMS5 of ECU manufacturer Gesas
  - avoid any multiple inheritance in IsoAgLib
  - change HAL modules for PC-RTE so that they read the RTE server name from standard environment variable RTE_HOST
  - class names match now a strict camel case pattern (e.g. ISOMonitor_c renamed to IsoMonitor_c)
  - bug fix: replacement of the first item in VT-Client CommandQueue for replacement corrected
  - bug fix: devpropertyhandler corrected (incoming command-responses on ChangeDesignator)
  - new WorkingSet announce strategy: no more automatic announce on Address-Claim
  - can_server now with socket interface (Windows and Linux) and message queues (Linux only) with common CAN device extensions (PEAK can card, A1, Sontheim, ...)
  - Central CAN FIFO, so that the BIOS/OS doesn't need to provide individual FIFOs for each receive MsgObj
  - bug fix: use rc_isoTerminal.data().time() instead of HAL::getTime() in VtServerInstance
  - bug fix: VtServerInstance uses getLastProcessedCanPkgTime() now for detecting dropp offs

\subsection Change_N 2.0 (changes on maintained version)
  - rewritten and enhanced TimePosGps_c
  - use isUnspecified() for ISOName instead of 0xff
  - avoid STL container list method call, as at least std::list<T>::size() works very slow
  - use slist only for compiler, where its support is known: GCC and Tasking
  - introduce new target hardware: Dickey John standard platform
  - added OutputList to vt2iso and ISO_Terminal (will be there in VT Revision 3)
  - added SendPriority feature to can_server
  - new ISOName_c constructor with just DEV-CLASS as parameter
  - VtClientServerCommunication_c: fix for dequeueing commands if list is empty
  - add PWM handling on ESX Babyboard (from old IsoAgLib pre-2.0 tree) and some PWM/Digout_c HAL fixes
  - can_server: avoid sending of CAN messages when destination address is one of the local clients
  - vt2iso: enable live editing
  - bug fix in hal_can_interface: send buffer overflow during pool upload
  - inserted send/receive of Vt_Server Phase3 message Lock/Unlock Mask
  - handling of display activation message (iIsoTerminalObjectPool_c::eventDisplayActivation())
  - improved ISOMonitor_c::unifyIsoISOName() when there's a conflict on the ISOBUS
  - fixed endless loop in xgpl_src/IsoAgLib/util/impl/util_funcs.h's convertLittleEndi...
  - fixed FASTPACKET sendIsoFastPacketBroadcast
  - changed ISONameUnspecified from static to ISONameUnspecified() function (problem with static (singleton) classes)
  - buffering of targetfilestreaminput to speed up byte by byte reading (activate with USE_BUFFERED_READ define)
  - bug fix in ISOMonitor_c::sendRequestForClaimedAddress() to ensure correct timing in address claim sequence
  - bug fix in ISOTerminal_c::init() to avoid problems (not seen VTs) when ISOTerminal_c::init() is being called too late
  - fix wrong handling of std::list<T>::erase() in canio_c.cpp
  - fixed TIMEOUT values in vtclientservercommunication_c.cpp
  - fix scheduler handling when a task got into negative area
  - Added feature for MultiReceive to set the time-period to 5000 when nothing to do

xgpl_src/IsoAgLib/hal/config.h
/***************************************************************************
                          config.h - include dependent on used target (defined
													           in Application_Config/isoaglib_config.h)
																		 the suitable HAL specific header for
																		 central configuration settings
                             -------------------
    begin                : Sun Mar 09 2003
    copyright            : (C) 2003 - 2004 Dipl.-Inform. Achim Spangler
    email                : a.spangler@osb-ag:de
    type                 : Header
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 * This file is part of the "IsoAgLib", an object oriented program library *
 * to serve as a software layer between application specific program and   *
 * communication protocol details. By providing simple function calls for  *
 * jobs like starting a measuring program for a process data value on a    *
 * remote ECU, the main program has not to deal with single CAN telegram   *
 * formatting. This way communication problems between ECU's which use     *
 * this library should be prevented.                                       *
 * Everybody and every company is invited to use this library to make a    *
 * working plug and play standard out of the printed protocol standard.    *
 *                                                                         *
 * Copyright (C) 1999 - 2004 Dipl.-Inform. Achim Spangler                  *
 *                                                                         *
 * The IsoAgLib is free software; you can redistribute it and/or modify it *
 * under the terms of the GNU General Public License as published          *
 * by the Free Software Foundation; either version 2 of the License, or    *
 * (at your option) any later version.                                     *
 *                                                                         *
 * This library is distributed in the hope that it will be useful, but     *
 * WITHOUT ANY WARRANTY; without even the implied warranty of              *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU       *
 * General Public License for more details.                                *
 *                                                                         *
 * You should have received a copy of the GNU General Public License       *
 * along with IsoAgLib; if not, write to the Free Software Foundation,     *
 * Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA           *
 *                                                                         *
 * As a special exception, if other files instantiate templates or use     *
 * macros or inline functions from this file, or you compile this file and *
 * link it with other works to produce a work based on this file, this file*
 * does not by itself cause the resulting work to be covered by the GNU    *
 * General Public License. However the source code for this file must still*
 * be made available in accordance with section (3) of the                 *
 * GNU General Public License.                                             *
 *                                                                         *
 * This exception does not invalidate any other reasons why a work based on*
 * this file might be covered by the GNU General Public License.           *
 *                                                                         *
 * Alternative licenses for IsoAgLib may be arranged by contacting         *
 * the main author Achim Spangler by a.spangler@osb-ag:de                  *
 ***************************************************************************/

/* ************************************************************ */
/** \file IsoAgLib/hal/config.h
  * include dependent on used target (defined in
	  Application_Config/isoaglib_config.h) the suitable HAL
		specific header for central configuration settings.
*/
/* ************************************************************ */
#ifndef _HAL_INDEPENDEND_CONFIG_H_
#define _HAL_INDEPENDEND_CONFIG_H_

// include interface aplication relevant configuration settings
#include <Application_Config/isoaglib_config.h>
#include <IsoAgLib/util/compiler_adaptation.h>

#define HAL_CAN_BITRATE_LIST {10, 20, 40, 50, 100, 125, 250, 500, 1000}
#define HAL_CAN_BITRATE_CNT 9
/** index for "LastMsgObj" where all messages are stored, which map through the special lastMsgMask */
#define HAL_CAN_LAST_MSG_OBJ_NR 14
#define HAL_CAN_MIN_SEND_OBJ 0
#define HAL_CAN_MAX_SEND_OBJ 13
// the send msg objects take the lowest possible msg ob numbers
// ==> take 2 as min msg obj nr
#define HAL_CAN_MIN_REC_OBJ 1
#define HAL_CAN_MAX_REC_OBJ 14

/// define list of allowed speed settings
#define HAL_RS232_BAUDRATE_LIST {75, 600, 1200, 2400, 4800, 9600, 19200, 38400, 57600}
#define HAL_RS232_BITRATE_CNT 9                                          

// now include dependent on used target the suitable header
#if defined(SYSTEM_PC)
	#include "pc/config.h"
#elif defined(SYSTEM_PC_MFC)
	#include "pc_mfc/config.h"
#elif defined(SYSTEM_ESX)
	#include "esx/config.h"
#elif defined(SYSTEM_A2)
	#include "a2/config.h"
#elif defined(SYSTEM_ESXu)
	#include "esxu/config.h"
#elif defined(SYSTEM_C2C)
	#include "c2c/config.h"
#elif defined(SYSTEM_DJ1)
	#include "Dj1/config.h"
#elif defined(SYSTEM_IMI)
	#include "imi/config.h"
#elif defined(SYSTEM_PM167)
	#include "pm167/config.h"
#elif defined(SYSTEM_MITRON167)
	#include "mitron167/config.h"
#elif defined(SYSTEM_AMS5)
	#include "ams5/config.h"
#endif


#endif
